src/workerd/api/web-socket.c++

```diff
@@ -185,8 +185,10 @@ jsg::Ref<WebSocket> WebSocket::constructor(jsg::Lock& js,
 
   // Check if we have a valid URL
   constexpr auto wsErr = "WebSocket Constructor: "_kj;
-  kj::Url urlRecord =
-      JSG_REQUIRE_NONNULL(kj::Url::tryParse(url), DOMSyntaxError, wsErr, "The url is invalid.");
+  // Any authority state (username and/or password) members should be removed.
+  // Users should use Authorization header for this purpose.
+  kj::Url urlRecord = JSG_REQUIRE_NONNULL(kj::Url::tryParse(url, kj::Url::HTTP_PROXY_REQUEST),
```

Comment:
Should we also have 
```suggestion
  auto urlOptions = kj::Url::Options{.percentDecode = false, .allowEmpty = true};
  kj::Url urlRecord = JSG_REQUIRE_NONNULL(kj::Url::tryParse(url, kj::Url::HTTP_PROXY_REQUEST, urlOptions),
```
here?

Reply:
In reality, allowEmpty should not make a difference but for the sake of completeness with fetch, yes we should add.

Although, we should percent decode. I don't see any harm in doing it?